telemetrylib/SQL.cpp

```diff
@@ -0,0 +1,129 @@
+//
+// Created by Mingcan Li on 2/3/23.
+//
+
+#include "DTI.h"
+#include <thread>
+#include <unistd.h>
+
+class SQL : public DTI {
+public:
+    SQL(QString tableToCreate) {
+        this->tableToCreate = tableToCreate;
+
+        restclient = new QNetworkAccessManager();
+
+        t = new std::thread(&SQL::checkConnection, this);
+        t->detach();
+    }
+
+    ~SQL() {
+        finish = true;
+        t->join();
+    }
+
+    void sendData(QByteArray bytes) override {
+        auto curr_msec = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+
+        QUrl myurl;
+        myurl.setScheme("http");
+        myurl.setHost("150.136.104.125"); // TODO
+        myurl.setPort(3000); // TODO
+        myurl.setPath("/add-data"); // TODO
+        myurl.setQuery("table-name=" + tableName + "&dataset-time=" + QString::fromStdString(std::to_string(curr_msec)));
+
+        //QNetworkRequest request;
+        request.setUrl(myurl);
+        request.setHeader(QNetworkRequest::ContentTypeHeader, QVariant("arraybuffer")); // TODO Try "blob" for content type as well
```

Comment:
```suggestion
        myurl.setHost("150.136.104.125");
        myurl.setPort(3000);
        myurl.setPath("/add-data");
        myurl.setQuery("table-name=" + tableName + "&dataset-time=" + QString::fromStdString(std::to_string(curr_msec)));

        //QNetworkRequest request;
        request.setUrl(myurl);
        request.setHeader(QNetworkRequest::ContentTypeHeader, QVariant("arraybuffer"));
```
Again, on me, but the TODOs can be removed